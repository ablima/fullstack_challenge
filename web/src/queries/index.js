import gql from "graphql-tag";

export const GET_PRODUCTS = gql`
  query Products($id: Int, $name: String, $categoryId: Int){
    getProducts(id: $id, name: $name, categoryId: $categoryId){
      id
      name
      description
      qnt
      price
      thumbnails
    }
  }
`

export const ADD_PRODUCT_TO_CART = gql`
  mutation Product($id: Int){
    addProductToCart(id: $id){
      id
    }
  }
`;

export const REMOVE_PRODUCT_FROM_CART = gql`
  mutation Product($id: Int){
    removeProductFromCart(id: $id){
      id
    }
  }
`;

export const GET_CATEGORIES = gql`
  query {
    getCategories{
      id
      name
    }
  }
`;

export const CREATE_ORDER = gql`  
  mutation Order($order: NewOrder){
    createOrder(order: $order){
      id
    }
  }
`;