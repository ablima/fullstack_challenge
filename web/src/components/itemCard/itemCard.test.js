import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import { ADD_PRODUCT_TO_CART } from '../../queries';
import ItemCard from './index';

const mocks = [
  {
    request: {
      query: ADD_PRODUCT_TO_CART,
      variables: {
        id: 17,
        qnt: 1
      }
    },
    result: {}
  }
];

const product = {
  name: "Produto",
  description: {},
  qnt: 5,
  price: 100,
  categoryId: 1,
  thumbnails: "images/celular1-1.jpg"
}

test('Check if render is correct', () => {
  const component = renderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ItemCard product={product} />
    </MockedProvider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});