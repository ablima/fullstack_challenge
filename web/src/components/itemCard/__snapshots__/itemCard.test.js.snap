// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if render is correct 1`] = `
<div
  className="card"
>
  <div
    className="cardInfo"
    id="cardButton"
    onClick={[Function]}
  >
    <div>
      <img
        src="http://localhost:3010/static/images/celular1-1.jpg"
      />
      <h3>
        Produto
      </h3>
      <h2>
        R$Â 100,00
      </h2>
    </div>
  </div>
  <div
    className="addButton"
    id="addButton"
    onClick={[Function]}
  >
    Adicionar ao carrinho
  </div>
</div>
`;
